nextcord/message.py

```diff
@@ -543,6 +545,56 @@ def flatten_handlers(cls):
     return cls
 
 
+class MessageInteraction:
+    """Represents a message's interaction data.
+
+    A message's interaction data is a property of a message when the message
+    is a response to an interaction from any bot.
+
+    .. versionadded:: 2.0
+
+    Attributes
+    -----------
+    data: :class:`dict`
+        The raw data from the interaction.
+    id: :class:`int`
+        The interaction's ID.
+    type: :class:`InteractionType`
+        The interaction type.
+    name: :class:`str`
+        The name of the application command.
+    user: Union[:class:`User`, :class:`Member`]
+        The :class:`User` who invoked the interaction or :class:`Member` if the interaction
+        occurred in a guild.
+    """
+
+    __slots__ = (
+        "_state",
+        "data",
+        "id",
+        "type",
+        "name",
+        "user",
+    )
+
+    def __init__(
+        self, *, data: MessageInteractionPayload, guild: Optional[Guild], state: ConnectionState
+    ):
+        self._state: ConnectionState = state
+
+        self.data: MessageInteractionPayload = data
+        self.id: int = int(data["id"])
+        self.type: int = data["type"]
+        self.name: str = data["name"]
+        if "member" in data and guild is not None:
+            self.user = Member(state=self._state, guild=guild, data={**data["member"], "user": data["user"]})  # type: ignore
+        else:
+            self.user = User(state=self._state, data=data["user"])
```

Comment:
This actually isn't working since User is only imported under TYPE_CHECKING. Another way to do this is
```suggestion
            self.user = self._state.create_user(data=data["user"])
```

Reply:
I originally had it as '_state.store_user', which does the same but also caches, both should work for just the data tho, 🤔